src/Templates/AdminPanelTemplate/Bit.AdminPanelTemplate/Api/Controllers/CategoryController.cs

```diff
@@ -0,0 +1,99 @@
+﻿using AdminPanelTemplate.Api.Models.Categories;
+using AdminPanelTemplate.Shared.Dtos.Categories;
+using AdminPanelTemplate.Api.Extensions;
+
+namespace AdminPanelTemplate.Api.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public partial class CategoryController : ControllerBase
+{
+    [AutoInject] private AppDbContext _dbContext = default!;
+
+    [AutoInject] private IMapper _mapper = default!;
+
+    [HttpGet, EnableQuery]
+    public IQueryable<CategoryDto> Get(CancellationToken cancellationToken)
+    {
+        return _dbContext.Categories
+            .ProjectTo<CategoryDto>(_mapper.ConfigurationProvider, cancellationToken);
+    }
+
+    [HttpGet("{id:int}")]
+    public async Task<CategoryDto> Get(int id, CancellationToken cancellationToken)
+    {
+        var category = await Get(cancellationToken).FirstOrDefaultAsync(t => t.Id == id, cancellationToken);
+
+        if (category is null)
+            throw new ResourceNotFoundException(nameof(ErrorStrings.CategoryCouldNotBeFound));
+
+        return category;
+    }
+
+    [HttpPost("GetPagedCategories")]
+    public async Task<PagedResultDto<CategoryDto>> GetPagedCategoriesAsync(PagedInputDto input, CancellationToken cancellationToken)
+    {
+        var query = Get(cancellationToken);
+
+        var total = query.Count();
+
+        var filteredQuery = query
+                            .WhereIf(!string.IsNullOrWhiteSpace(input.Filter), _ => _.Name!.Contains(input.Filter));
+
+        var orderedQuery = (input.SortBy, input.SortAscending) switch
+        {
+            (nameof(Category.Name), true) => filteredQuery.OrderBy(c => c.Name),
+            (nameof(Category.Name), false) => filteredQuery.OrderByDescending(c => c.Name),
+            _ => filteredQuery.OrderBy(c => c.Id)
+        };
+
+        var pageResult = await orderedQuery
+                         .PageBy(input.Skip, input.MaxResultCount)
+                         .ToListAsync();
```

Comment:
Pass cancellationToken here

Reply:
It's passed in the first line of the method
var query = Get(cancellationToken);